packages/shared/src/lib/core/layer-2/actions/fetchL2BalanceForAccount.ts

```diff
@@ -84,11 +84,11 @@ async function getLayer2NativeTokenBalancesForAddress(
         const contract = chain.getContract(ContractType.IscMagic, ISC_MAGIC_CONTRACT_ADDRESS)
         const nativeTokenResult = (await contract.methods
             .callView(accountsCoreContract, getBalanceFunc, parameters)
-            .call()) as { items: KeyValue<number>[] }
+            .call()) as { items: KeyValue<string>[] }
 
         const nativeTokens = nativeTokenResult.items.map((item) => ({
             tokenId: item.key,
-            balance: Math.floor(Number(item.value)),
+            balance: BigInt(item.value),
```

Comment:
is the new convertor function safer here for type any?

Reply:
BigInt is able to convert hex encoded strings, the new convertor function is useful, because it takes care in case of BigIntLike types that can be of type buffer. The response should not be returned by the smart contract in buffer format.